adal/src/main/java/com/microsoft/aad/adal/TokenCacheItem.java

```diff
@@ -122,122 +129,259 @@ public static TokenCacheItem createRegularTokenCacheItem(final String authority,
      * Will not store clientId, resource and AT. 
+     * 
+     * @param authority required authority identifier.
+     * @param authResult required authentication result to create FRRT refresh token cache item.
+     *                   
+     * @return TokenCacheItem
      */
     public static TokenCacheItem createFRRTTokenCacheItem(final String authority, final AuthenticationResult authResult) {
         return new TokenCacheItem(authority, authResult);
     }
 
+    /**
+     * Get the user information.
+     * 
+     * @return UserInfo object.
+     */
     public UserInfo getUserInfo() {
         return mUserInfo;
     }
 
+    /**
+     * Set the user information.
+     * 
+     * @param info UserInfo object which contains user information.
+     */
     public void setUserInfo(UserInfo info) {
         this.mUserInfo = info;
     }
 
+    /**
+     * Get the resource.
+     * 
+     * @return resource String.
+     */
     public String getResource() {
         return mResource;
     }
 
+    /**
+     * Set the resource.
+     * 
+     * @param resource resource identifier.
+     */
     public void setResource(String resource) {
         this.mResource = resource;
     }
 
+    /**
+     * Get the authority.
+     * 
+     * @return authority url string.
+     */
     public String getAuthority() {
         return mAuthority;
     }
 
+    /**
+     * Set the authority.
+     * 
+     * @param authority String authority url.
+     */
     public void setAuthority(String authority) {
         this.mAuthority = authority;
     }
 
+    /**
+     * Get the client identifier.
+     * 
+     * @return client identifier string.
+     */
     public String getClientId() {
         return mClientId;
     }
 
+    /**
+     * Set the client identifier.
+     * 
+     * @param clientId client identifier string.
+     */
     public void setClientId(String clientId) {
         this.mClientId = clientId;
     }
 
+    /**
+     * Get the access token.
+     * 
+     * @return the access token string.
+     */
     public String getAccessToken() {
         return mAccessToken;
     }
 
+    /**
+     * Set the access token string.
+     * 
+     * @param accessToken the access token string.
+     */
     public void setAccessToken(String accessToken) {
         this.mAccessToken = accessToken;
     }
 
+    /**
+     * Get the refresh token string.
+     * 
+     * @return the refresh token string.
+     */
     public String getRefreshToken() {
         return mRefreshtoken;
     }
 
+    /**
+     * Set the fresh token string.
+     * 
+     * @param refreshToken the refresh token string.
+     */
     public void setRefreshToken(String refreshToken) {
         this.mRefreshtoken = refreshToken;
     }
 
+    /**
+     * Get the expire date.
+     * 
+     * @return the time the token get expired.
+     */
     public Date getExpiresOn() {
-        return mExpiresOn;
+        return mExpiresOn != null ? new Date(mExpiresOn.getTime()) : null;
```

Comment:
I'm not sure if we should suppress it. As it described in the report. The EI_EXPOSE_REP is warning us that we should not expose the internal references of the object to the outside world so that they can not be modified by a malicious/erroneous code. Let's discuss about it. :)